src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -102,6 +103,26 @@ public MgmtOutputLibrary(CodeModel codeModel, BuildContext<MgmtOutputLibrary> co
             DecorateOperationSets();
         }
 
+        private void UpdateSubscriptionIdForAllResource(CodeModel codeModel)
+        {
+            foreach (var operationGroup in codeModel.OperationGroups)
+            {
+                foreach (var op in operationGroup.Operations)
+                {
+                    var newpParams = op.Parameters.ToList();
+                    for (int i = 0; i < newpParams.Count; i++)
+                    {
+                        //updater the first subscriptionId to be 'method'
+                        if (newpParams[i].Language.Default.Name.Equals("subscriptionId", StringComparison.InvariantCultureIgnoreCase))
+                        {
+                            newpParams[i].Implementation = ImplementationLocation.Method;
+                            break;
+                        }
+                    }
```

Comment:
Since the class `Parameter` is a reference type, I believe it is valid if we convert this iteration into a `foreach` loop like this:
```suggestion
                    for (var p in op.Parameters)
                    {
                        //updater the first subscriptionId to be 'method'
                        if (p.Language.Default.Name.Equals("subscriptionId", StringComparison.InvariantCultureIgnoreCase))
                        {
                            p.Implementation = ImplementationLocation.Method;
                            break;
                        }
                    }
```